server/bin/populate_plan_area_changes.js

```diff
@@ -7,21 +7,36 @@ const populatePlanAreaChanges = async () => {
 			qb.where('areaChanges', '!=', '[[]]').select('id','areaChanges');
 		})
 		.fetchAll();
-		areaChanges = result.models;
+		const areaChanges = result.models;
 		let howMuchLeft = areaChanges.length;
+
+		console.log('starting to populate plan_area_changes table');
 		for (let counter = 0; counter < areaChanges.length; counter++) {
 			const planId = areaChanges[counter].attributes.id;
 			const rawAreaChangesString = areaChanges[counter].attributes.areaChanges;
-			console.log(`${howMuchLeft} plans left`);
+
+			await PlanAreaChangesController.refreshPlanAreaChanges(planId, rawAreaChangesString);
 			howMuchLeft--;
-			await PlanAreaChangesController.refreshPlanAreaChanges(planId, rawAreaChangesString)
-			
-		}		
+
+			if (howMuchLeft % 100 === 0 && (howMuchLeft !== 0)) {
+				console.log(`${howMuchLeft} plans left`);
+			}
+
+		}
+
+		console.log('done populating plan_area_changes table');
 	}
 	catch (err) {
 		console.log(err);
 	}
 
 };
 
-populatePlanAreaChanges().then(() => console.log('done populating plan area changes'));
+
+
+if (require.main === module) {
```

Comment:
What is this condition for?

Reply:
This file is intended to be a runnable file and an importable file. This if is checking if this file ia a main. If it's a main, we should execute the function. If it's loaded as an import, we shouldn't execute the function.

Reply:
Why don't we separate it into 2?

Reply:
Why? It's the same code.

Reply:
But two different purposes? The runnable (executable) file will *only * call the 1 function in the logic file

Reply:
Fixed by splitting this file into a code file and a runnable, and put them in a folder inside /bin.